src/express/commands/start.js

```diff
@@ -32,28 +32,42 @@ async function installRequiredSoftwareOnRemoteMachines(ips, devnetType) {
 
     let ipsArray = splitToArray(ips)
     let borUsers = splitToArray(doc['devnetBorUsers'].toString())
-
     let user, ip
+    let nodeIps = []
+    let isHostMap = new Map()
 
     for (let i = 0; i < ipsArray.length; i++) {
-
-        i === 0 ? user = `${doc['ethHostUser']}` : `${borUsers[i]}`
+        i === 0 ? user = `${doc['ethHostUser']}` : user =`${borUsers[i]}`
         ip = `${user}@${ipsArray[i]}`
+        nodeIps.push(ip)
 
-        await configureCertAndPermissions(user, ip)
-        await installCommonPackages(user, ip)
+        i === 0 ? isHostMap.set(ip, true) : isHostMap.set(ip, false)
+    }
 
-        if (i === 0) {
+    let deps = nodeIps.map(async(ip) => {
+        await configureCertAndPermissions(ip)
+        await installCommonPackages(ip)
+
+        if (isHostMap.get(ip)) {
+            // Install Host dependencies
             await installHostSpecificPackages(ip)
 
             if (process.env.TF_VAR_DOCKERIZED === 'yes') {
-                await installDocker(ip, user)
+                await installDocker(ip)
             }
         }
-    }
+
+    })
+
+    await Promise.all(deps)
 }
 
-async function configureCertAndPermissions(user, ip) {
+async function configureCertAndPermissions(ip) {
```

Comment:
can you pass the user here as before, to avoid a re-split?

Reply:
Done: [a2b9787](https://github.com/maticnetwork/matic-cli/pull/82/commits/a2b978715cd75135ede2db6185ff6f395f4e8c15)